src/taco-cli/cli/utils/settings.ts

```diff
@@ -87,7 +87,7 @@ class Settings {
             Settings.determinePlatformsFromOptions(options)
         ]).spread<Settings.IPlatformWithLocation[]>(function (supportedPlatforms: CordovaHelper.IDictionary<any>, platforms: Settings.IPlatformWithLocation[]): Settings.IPlatformWithLocation[] {
             var filteredPlatforms = platforms.filter(function (platform: Settings.IPlatformWithLocation): boolean {
-                var supported = !supportedPlatforms || platform.platform in supportedPlatforms;
+                var supported = !supportedPlatforms || platform.platform in supportedPlatforms || platform.location === Settings.BuildLocationType.Remote;
```

Comment:
In determinePlatformsFromOptions we may end up with non-supported platforms. For example, you can "taco platform add ios" on windows, and then "taco build" will pick it up as part of 'all installed platforms'. Similarly someone could explicitly state "taco build ios --local" on windows, where ios is not a supported platform, and determinePlatformsFromOptions will include it as a local platform.

I didn't want to collapse both the determinePlatform and determinePlatformsFromOptions into the one function, since telemetry wants to see what the user asked for, not just what we ended up doing.